File: assgn2/TestDeque.java
public class TestDeque {
    private Deque<Integer> d;

    public TestDeque() {
        d = new Deque<Integer>();
    }

    public void addAndRemove() {
        int i;
        int[] a = new int[10];
        for (i = 0; i < 10; i++) {
            int x = StdRandom.uniform(0, 100);
            a[i] = x;
            d.addLast(x);
        }

        assert (d.size() == 10);
        for (i = 0; i < 10; i++) {
            int x = d.removeLast();
            assert x == a[10-i-1];
        }

        for (i = 0; i < 10; i++) {
            int x = StdRandom.uniform(0, 100);
            a[i] = x;
            d.addFirst(x);
        }

        for (i = 0; i < 10; i++) {
            int x = d.removeFirst();
            assert x == a[10-i-1];
        }

        for (i = 0; i < 10; i++) {
            int x = StdRandom.uniform(0, 100);
            a[i] = x;
            d.addLast(x);
        }

        for (i = 0; i < 10; i++) {
            int x = d.removeFirst();
            assert x == a[i];
        }

        for (i = 0; i < 10; i++) {
            int x = StdRandom.uniform(0, 100);
            a[i] = x;
            d.addFirst(x);
        }

        for (i = 0; i < 10; i++) {
            int x = d.removeLast();
            assert x == a[i];
        }

        assert d.size() == 0;
    }

    public void iteration() {
        int i;
        int[] a = new int[10];

        for (i = 0; i < 10; i++) {
            int x = StdRandom.uniform(0, 100);
            a[i] = x;
            d.addLast(x);
        }


        i = 0;
        for (int x: d)
            assert x == a[i++];
    }

    public static void main(String[] args) {
        TestDeque td = new TestDeque();
        StdOut.println("Testing addition and removal...");
        td.addAndRemove();
        StdOut.println("Testing iteration...");
        td.iteration();
    }
}

File: python/queues/test_deque.py
from deque import Deque

def test_addition():
    print "Testing addition..."
    dq = Deque()
    assert dq.size() == 0

    for i in range(10):
        dq.add_last(i)

    assert dq.size() == 10

    for i in range(10):
        x = dq.remove_first()
        assert x == i

    assert dq.size() == 0

    for i in range(10):
        x = dq.add_first(i)

    for i in range(10):
        x = dq.remove_last()
        assert x == i

def test_iteration():
    print "Testing iteration..."
    dq = Deque()
    for i in range(10):
        dq.add_last(i)

    a = range(10)
    index = 0
    for i in dq:
        assert i == a[index]
        index += 1

if __name__ == "__main__":
    test_addition()
    test_iteration()
